components/MMU/MMUImpl.cpp

```diff
@@ -140,10 +141,11 @@ class FLEXUS_COMPONENT(MMU) {
     }
 
     std::pair<bool, PhysicalMemoryAddress> lookUp(const VirtualMemoryAddress &anAddress) {
+      VirtualMemoryAddress anAddressAligned(anAddress & PAGEMASK);
```

Comment:
Wouldn't it be better to do this by looking up the currently supported Translation Granule size in the system register ID_AA64MMFR0_EL1?

AFAIK I wrote some basic support for multiple translation granule sizes in "TranslationGranules.cpp" a while ago. Not sure if those have been updated in the recent past however, or that support got removed somehow?

Reply:
This functionality overall corresponds to using different page sizes, which will require adding and testing appropriate functionality for both TLB lookups and page table walks.
I am not sure if a simple change will do it.
So, I would rather stick with 4KB pages for now.

Reply:
Yes, I'm aware it means a different page size. However I think instead of hardcoding this with a bitmask, we should lookup the translation granules module, and if the config registers show a different translation granule size, assert error. This means we will immediately know the source (if there is a future problem), as well as have the support already there if anyone implements larger TGs in the future. Thoughts?

Reply:
I can put asserts during MMU initialization to verify if the TG modules result in 4KB pages.
And then use the retrieved value as the page size.
Here: https://github.com/parsa-epfl/flexus/pull/14/commits/f880cbe539b4605fce8ec915cf16d557e40df5ca

Reply:
Nice 👍